### ISSUE TITLE
Can not authenticate LDAP users

### ISSUE THREAD

**Issue by v1r0x** [jsxc/jsxc#171]
I installed LDAP on my server and everything works fine except the chat.

The
`$ret = $ocUser->userExists($argv[2]);`
command returns 0 instead of 1. I tried to get all users and I get all users except the LDAP users.
In another App I get all users, including the LDAP users.

**Comment by sualko** [comment]
Do you use internal and ldap authentication in parallel?

First of all <code>argv[2]</code> is the node part of your jid (<code><JID> ::= [<node>"@"]<domain>["/"<resource>]</code>)  and <code>userExists($uid)</code> takes the uid to search for your user. So I guess your ldap uid is different from the node. Is this right?

**Comment by v1r0x** [comment]
Yes, currently I use both.

I configured the LDAP owncloud add on to use the user name as uid instead of the default hash (or whatever). So the LDAP uid is the same as the owncloud uid (for users which have both LDAP and OC uid the addon adds four digits to the LDAP uid).
If I use something like this in another app (in my example I added it to the polls app):
```php
print_r(OCP\User::getUsers());
```
I get all users, including LDAP users.
But if I use the same code in the `auth_oc_user.php` I only get the OC users and not the LDAP users.
